src/text_files_splitter/text_files_splitter.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+
+import os
+
+
+class ListMetrics:
+    B = ['B', 0]
+    KB = ['KB', 1]
+    MB = ['MB', 2]
+    GB = ['GB', 3]
+    TB = ['TB', 4]
+    PB = ['PB', 5]
+
+
+class BytesMultiples:
+    DECIMAL = 1000
+    BINARY = 1024
+
+
+def remove_file(file_name):
+    os.remove(file_name)
+
+
+def get_file_size_bytes(file_name):
+    return os.path.getsize(file_name)
+
+
+def get_file_size_metric(file_bytes, metric=ListMetrics.B, bytes_multiples=BytesMultiples.DECIMAL):
+
+    total_divisions = metric[1]
+    file_size_metric = file_bytes
+
+    while total_divisions:
+        file_size_metric /= bytes_multiples
+        total_divisions -= 1
+
+    return file_size_metric
+
+
+def get_size_to_metric(size, metric):
+
+    while metric:
+        size *= BytesMultiples.DECIMAL
+        metric -= 1
+
+    return size
+
+
+def get_total_file_splits(file_bytes, file_size_limit):
+
+    splits = file_bytes / file_size_limit
+    return int(splits) if splits.is_integer() else int(splits) + 1
+
+
+def delete_file_last_row(file):
+    file_rows_exceeded_size = open(file, 'r+').readlines()
+    file_rows_exceeded_size.pop()
+
+    file_finished = open(file, 'w+')
+    file_finished.writelines(file_rows_exceeded_size)
+
+
+def get_file_name_splitted(file_suffix, split_count):
+    return '{}_{}.csv'.format(file_suffix, split_count)
+
+
+def create_file_splitted(output_directory, file_splitted_name):
+    file_path = os.path.join(output_directory, file_splitted_name)
+    return open(file_path, 'w')
+
+
+def close_file(file_object):
+    file_object.close()
+
+
+class TextFilesSplitter():
+    def __init__(self, source_file, file_name, max_size_file, file_type='csv', metric_units=ListMetrics.B, output_directory='.', include_header=True, file_suffix=None):
```

Comment:
Why do we need source_file and file_name to be separate? We can receive all in one and then split it if we need to.

Reply:
ddd